contracts/LosslessControllerV3.sol

```diff
@@ -381,36 +380,59 @@ contract LosslessControllerV3 is Initializable, ContextUpgradeable, PausableUpgr
     /// @param recipient Address to add the locks
     function _enqueueLockedFunds(ReceiveCheckpoint memory checkpoint, address recipient) private {
         LocksQueue storage queue;
-        queue = tokenScopedLockedFunds[msg.sender].queue[recipient];
-        if (queue.lockedFunds[queue.last].timestamp == checkpoint.timestamp) {
+        queue = tokenScopedLockedFunds[ILERC20(msg.sender)].queue[recipient];
+
+        if (queue.lockedFunds[queue.last].timestamp < checkpoint.timestamp) {
```

Comment:
save queue.last in local variable and reuse. CVF-114

Reply:
Fixed on new commit 4bec697